.github/actions/setup-terraform/action.yaml

```diff
@@ -0,0 +1,87 @@
+name: Setup Terraform
+description: Install and configure Terraform and AWS for the correct environment
+inputs:
+  role-to-assume:
+    description: AWS IAM role to assume when running Terraform operations.
+    required: true
+  aws-account-id:
+    description: AWS account ID to use to create resources.
+    required: true
+  batch-container-image-name:
+    description: The name of the container image to use for the Batch job.
+    required: true
```

Comment:
This input variable is indicative of the behavior that seems trickiest to imagine factoring out: Variables that are specific to the Terraform workflow in question. We might be able to get away with a generic variable like `terraform-vars` that accepts a comma-delimited string of Terraform variables:

```yaml
with:
  terraform-vars: "batch-container-image-name=foobar"
```

At this point it's all a bit speculative, however, so I'm punting on it unless/until we decide to factor this logic out into a cross-repo shared action.